lib/publiccloud/gce.pm

```diff
@@ -170,14 +170,17 @@ sub cleanup {
 
     my $region = $self->{provider_client}->{region};
     my $project = $self->{provider_client}->{project_id};
-    my $instance_id = $self->get_terraform_output(".vm_name.value[0]");
+    my $query = check_var('PUBLIC_CLOUD_SLES4SAP', 1) ? ".hana_name.value[0]" : ".vm_name.value[0]";
```

Comment:
This one introduce a dependency that is very difficult to maintain. This query depend now to code that leave in https://github.com/SUSE/qe-sap-deployment/blob/dcfe4b2b894c95f611eae0f35652644a62f78be8/terraform/gcp/outputs.tf#L35

I suggest not to have this dependency

Reply:
This specific case in `cleanup`, where that routine is called, is tailored on the data found during VR SAP tests pre-checks, to provide a valid output in this specific routine. 
But _SAP_ can call it everywhere needeed to get proper data entering any better fitting input query.
Calling `get_terraform_output` elsewhere you can choose **any** `terraform output` query based on data available and wanted.

Reply:
It was also to show SAP applicability: I will update the code to only use the normal p.c. default query `.vm_name.value[0]`